### PR TITLE
TT-7899 Attributes

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

- Added attributes abstraction for traces
- Added API semconv for Tyk

## Motivation and Context

https://tyktech.atlassian.net/browse/TT-7899
## Test Coverage For This Change

https://tyktech.atlassian.net/browse/TT-7899
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] I have [reviewed the guidelines](./CONTRIBUTING.md) for contributing to this repository.
- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
     - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
    - [ ] `gofmt -s -w .`
    - [ ] `go vet ./...`

### CODE REVIEW

File: semconv/v1.0.0/tykapis.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+// Auth related attributes
+const (
+	// equivalent to Tyk Pump APIKey(api_key)
+	TykAPIKeyKey = attribute.Key(TykAPIPrefix + "apikey")
+
+	// equivalent to Tyk Pump Alias(alias)
+	TykAPIKeyAliasKey = attribute.Key(TykAPIPrefix + "apikey.alias")
+
+	// equivalent to Tyk Pump OauthID(oauth_id)
+	TykOauthIDKey = attribute.Key(TykAPIPrefix + "oauthid")
```
Comment (sredxny) [inline semconv/v1.0.0/tykapis.go:43]: do we need one for certs? for the mtls use-cases?

**Review by sredxny — APPROVED** [review]

File: trace/attributes.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	case []int:
+		return attribute.Key(key).IntSlice(v)
+	case []int64:
+		return attribute.Key(key).Int64Slice(v)
+	case []float64:
+		return attribute.Key(key).Float64Slice(v)
+	case fmt.Stringer:
+		return attribute.Key(key).String(v.String())
+	default:
+		return attribute.Key(key).String(fmt.Sprint(v))
```
Comment (mativm02) [inline trace/attributes.go:55]: why are we defaulting to string here? isn't there a `Interface()` method? 🤔

Comment (tbuchaillot) [reply trace/attributes.go:55]: Nope - there's no Interface() method.

File: semconv/v1.0.0/tykapis.go (hunk 1 above)
Comment (tbuchaillot) [reply semconv/v1.0.0/tykapis.go:46]: for the moment nope - let's iterate later :)

**Pull request merged**